### PR TITLE
Update the libcxx (llvm project) to version 19.1.1

### PR DESCRIPTION
**Pull request by Noremos**
Update the libcxx custom package to remove the dependency on an old Python version

### CODE REVIEW

**Comment by Noremos** [comment]
The ARM build failed during the unit test compilation. It seems like it is necessary to update BOOST. It uses std::unary_function, which was deleted in C++17. I checked, and the updated BOOST version works. It would be nice to update BOOST, but it consists of over 2000 files. I can create another PR, but I'm not sure if all this is really needed by anyone besides me.

**Comment by aafemt** [comment]
If you are sure that this piece of BOOST cannot be eliminated from Firebird tree completely - try to set `_HAS_AUTO_PTR_ETC` to 1. According to the BOOST sources it should switch it to usage of own implementation of `unary_function`.

**Comment by asfernandes** [comment]
Doing boost update in https://github.com/FirebirdSQL/firebird/pull/8286

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -312,15 +312,6 @@ jobs:
...
-        run: |
-          brew uninstall --ignore-dependencies python@3.12
-          [ -e /Library/Frameworks/Python.framework ] && \
-            sudo mv /Library/Frameworks/Python.framework /Library/Frameworks/Python.framework.dont-look-at-this
-          brew install python@3.11
-          ln -s "$(brew --prefix python@3.11)/bin/python3" "$(brew --prefix python@3.11)/bin/python"
-
       - name: Cache - vcpkg
         uses: actions/cache@v4
         with:
```
Comment (asfernandes) [inline .github/workflows/main.yml:317]: Should also change cache key below, to something like `key: ${{ runner.os }}-vcpkg-1`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -312,27 +312,18 @@ jobs:
...
+          key: ${{ runner.os }}-vcpkg-fb6-0
           path: |
             $HOME/.cache/vcpkg/archives
 
       - name: Build
         run: |
           export LIBTOOLIZE=glibtoolize
           export LIBTOOL=glibtool
-          export PATH="$(brew --prefix python@3.11)/bin:$PATH"
+          export PATH="$(brew --prefix python3)/bin:$PATH"
```
Comment (asfernandes) [inline .github/workflows/main.yml:326]: Is this still necessary?

**Comment by Noremos** [comment]
I'm a little concerned about the “Cache not found for input keys” message in the cache action. Does it really work?

**Comment by asfernandes** [comment]
> I'm a little concerned about the “Cache not found for input keys” message in the cache action. Does it really work?

Seems it was not already working.
The path is invalid.

**Comment by Noremos** [comment]
I tried using a different approach for vcpkg dependency caching, described in https://learn.microsoft.com/en-us/vcpkg/consume/binary-caching-github-actions-cache. 
And it seems to be working: https://github.com/Noremos/firebird/commit/4b5eeb2f799d43ea379a34bc467c85c9525381f2

`Restored 3 package(s) from GitHub Actions Cache in 12 s.`

**Comment by asfernandes** [comment]
It's an experimental flag and is not clear for me what it use as cache key and how it will be in regard to multiple arch and branches.
But we may go with it and see.

**Comment by asfernandes** [comment]
I did some tests and it caches packages individually with their hashs. Seems good.
Could you incorporate it here?
Is it the only pendency for merge?

**Comment by Noremos** [comment]
> Could you incorporate it here?
Commied the change

> Is it the only pendency for merge?
Yes. If there are no problems with CI and there are no additional issues, then everything is ready

**Pull request merged**